pass/ProteusPass.cpp

```diff
@@ -738,37 +637,20 @@ class ProteusJitPassImpl {
     return true;
   }
 
-  void replaceWithJitLaunchKernel(Module &M, CallBase *LaunchKernelCall) {
-    GlobalVariable *ModuleUniqueId =
-        M.getGlobalVariable("__module_unique_id", true);
-    assert(ModuleUniqueId && "Expected ModuleUniqueId global to be defined");
-
-    if (!FatbinName)
-      FATAL_ERROR("replaceWithJitLaunchKernel not callable without EnableHIP "
-                  "or EnableCUDA set.");
-
-    GlobalVariable *FatbinWrapper = nullptr;
-    FatbinWrapper = M.getGlobalVariable(FatbinName, true);
-    if (!FatbinWrapper)
-      FATAL_ERROR(
-          "Expected hip fatbinary wrapper global, check "
-          "ENABLE_CUDA|ENABLE_HIP compilation flags for ProteusJitPass");
-
-    FunctionType *JitEntryFnTy = nullptr;
+  FunctionCallee getJitLaunchKernelFn(Module &M) {
+    FunctionType *JitLaunchKernelFnTy = nullptr;
```

Comment:
Since this is a refactor once more. I would add a TODO/FIXME note here and mention the ABI issue

Reply:
There is an issue #47